pkg/proc/core/linux_core.go

```diff
@@ -302,8 +375,8 @@ type LinuxPrPsInfo struct {
 	Args                 [80]uint8
 }
 
-// LinuxPrStatus is a copy of the prstatus kernel struct.
-type LinuxPrStatus struct {
+// LinuxPrStatusAMD64 is a copy of the prstatus kernel struct.
+type LinuxPrStatusAMD64 struct {
```

Comment:
It's a shame we have to dup these structs just because of the Reg value, tried to think of a way to prevent that using a custom binary unmarshaler, but I don't think it'll work. Oh well.

Reply:
@derekparker is this fine or you want some changes ? it's not clear

Reply:
> It's a shame we have to dup these structs just because of the Reg value, tried to think of a way to prevent that using a custom binary unmarshaler, but I don't think it'll work. Oh well.

Yes, how about move `proc/registers.go` into `proc/linutil/`, and use interface instead in packages. (not in this patch, code refactor next time)

Reply:
> @derekparker is this fine or you want some changes ? it's not clear

Sorry, no changes requested right now. I was just thinking out loud, apologies for not being clearer.